# Telephone Directory CRUD Operation

Telephone directory: Perform CRUD operation using mongodb and python.
You need to

Import necessary modules.
Perform CRUD operations to manipulate data in MongoDB. Create, retrieve,update, and delete (CRUD)
Create a database using attribute style on a MongoClient instance. 

Declare a variable db and assign the new database as an attribute of the client.

Create a collection.


For CRUD operation, create a directory which has fields like Name, Phone number, Place etc.,

Insert the record into the collection.

Make a query to find records you just created.

Modify the records, use the update_one() method. 

The update_one() method requires two arguments, query and update.

Delete the record, use delete_one() method. delete_one() requires a query parameter which specifies the document to delete.


In [1]:
# Importing the Modules
import pymongo
import json
from pymongo import MongoClient

In [2]:
myclient = pymongo.MongoClient("mongodb://127.0.0.1:27017/")

In [3]:
# Create a database using attribute style on a MongoClient instance. Declare a variable db and assign the new database as an attribute of the client.
db = myclient["Telephone"]

data = db.collection1
data.count_documents({}) ## just finding if any data is present

0

In [4]:
# For CRUD operation, create a directory which has fields like Name, Phone number, Place etc.,

directory = [
    {'telid':1,'name':'ajith','place':'gudivada','phonenumber':8956237845},
    {'telid':2,'name':'akash','place':'machilipatnam','phonenumber':4578122356},
    {'telid':3,'name':'manohar','place':'bhimavaram','phonenumber':4589562358},
    {'telid':4,'name':'ajay','place':'tanuku','phonenumber':9856231245},
    {'telid':5,'name':'dhanu','place':'vijayawada','phonenumber':7895451223},
    {'telid':6,'name':'lakshmi','place':'nellore','phonenumber':8745568585},
    {'telid':7,'name':'narayana','place':'nuzvid','phonenumber':7812456235},
    {'telid':8,'name':'ram','place':'moturu','phonenumber':7845962158},
    {'telid':9,'name':'lakshman','place':'tarigoppula','phonenumber':7895486215},
    {'telid':10,'name':'rosy','place':'eluru','phonenumber':9856487452}
]

In [5]:
directory_data = data.insert_many(directory)
print(directory_data.inserted_ids)

[ObjectId('635ead793af58a47440d2ce6'), ObjectId('635ead793af58a47440d2ce7'), ObjectId('635ead793af58a47440d2ce8'), ObjectId('635ead793af58a47440d2ce9'), ObjectId('635ead793af58a47440d2cea'), ObjectId('635ead793af58a47440d2ceb'), ObjectId('635ead793af58a47440d2cec'), ObjectId('635ead793af58a47440d2ced'), ObjectId('635ead793af58a47440d2cee'), ObjectId('635ead793af58a47440d2cef')]


In [6]:
# Insert the record into the collection.

y_data = {'telid':11,'name':'jersy','place':'ronaldo','phonenumber':9856487852}
directory_data1 = data.insert_one(y_data)

In [7]:
telephone_data = db.collection1.find()

for i in telephone_data:
    print(i)

{'_id': ObjectId('635ead793af58a47440d2ce6'), 'telid': 1, 'name': 'ajith', 'place': 'gudivada', 'phonenumber': 8956237845}
{'_id': ObjectId('635ead793af58a47440d2ce7'), 'telid': 2, 'name': 'akash', 'place': 'machilipatnam', 'phonenumber': 4578122356}
{'_id': ObjectId('635ead793af58a47440d2ce8'), 'telid': 3, 'name': 'manohar', 'place': 'bhimavaram', 'phonenumber': 4589562358}
{'_id': ObjectId('635ead793af58a47440d2ce9'), 'telid': 4, 'name': 'ajay', 'place': 'tanuku', 'phonenumber': 9856231245}
{'_id': ObjectId('635ead793af58a47440d2cea'), 'telid': 5, 'name': 'dhanu', 'place': 'vijayawada', 'phonenumber': 7895451223}
{'_id': ObjectId('635ead793af58a47440d2ceb'), 'telid': 6, 'name': 'lakshmi', 'place': 'nellore', 'phonenumber': 8745568585}
{'_id': ObjectId('635ead793af58a47440d2cec'), 'telid': 7, 'name': 'narayana', 'place': 'nuzvid', 'phonenumber': 7812456235}
{'_id': ObjectId('635ead793af58a47440d2ced'), 'telid': 8, 'name': 'ram', 'place': 'moturu', 'phonenumber': 7845962158}
{'_id': Ob

In [8]:
# Make a query to find records you just created.
yda = {'telid':9}
db.collection1.find_one(yda)

{'_id': ObjectId('635ead793af58a47440d2cee'),
 'telid': 9,
 'name': 'lakshman',
 'place': 'tarigoppula',
 'phonenumber': 7895486215}

In [9]:
myquery = {'place':'tanuku'}
mydat = db.collection1.find(myquery,{'telid':0,'place':0})
for x in mydat:
  print(x)

{'_id': ObjectId('635ead793af58a47440d2ce9'), 'name': 'ajay', 'phonenumber': 9856231245}


In [10]:
# Modify the records, use the update_one() method. 

myquery = {"name":"ram"}
newval = {"$set":{"name":"RaghuRamRaavi"}}
data.update_one(myquery,newval)
for x in data.find():
  print(x)

{'_id': ObjectId('635ead793af58a47440d2ce6'), 'telid': 1, 'name': 'ajith', 'place': 'gudivada', 'phonenumber': 8956237845}
{'_id': ObjectId('635ead793af58a47440d2ce7'), 'telid': 2, 'name': 'akash', 'place': 'machilipatnam', 'phonenumber': 4578122356}
{'_id': ObjectId('635ead793af58a47440d2ce8'), 'telid': 3, 'name': 'manohar', 'place': 'bhimavaram', 'phonenumber': 4589562358}
{'_id': ObjectId('635ead793af58a47440d2ce9'), 'telid': 4, 'name': 'ajay', 'place': 'tanuku', 'phonenumber': 9856231245}
{'_id': ObjectId('635ead793af58a47440d2cea'), 'telid': 5, 'name': 'dhanu', 'place': 'vijayawada', 'phonenumber': 7895451223}
{'_id': ObjectId('635ead793af58a47440d2ceb'), 'telid': 6, 'name': 'lakshmi', 'place': 'nellore', 'phonenumber': 8745568585}
{'_id': ObjectId('635ead793af58a47440d2cec'), 'telid': 7, 'name': 'narayana', 'place': 'nuzvid', 'phonenumber': 7812456235}
{'_id': ObjectId('635ead793af58a47440d2ced'), 'telid': 8, 'name': 'RaghuRamRaavi', 'place': 'moturu', 'phonenumber': 7845962158}


In [11]:
# Delete the record, use delete_one() method.

myquery = {'phonenumber':7895451223}
data.delete_one(myquery)
for x in data.find():
    print(x)

{'_id': ObjectId('635ead793af58a47440d2ce6'), 'telid': 1, 'name': 'ajith', 'place': 'gudivada', 'phonenumber': 8956237845}
{'_id': ObjectId('635ead793af58a47440d2ce7'), 'telid': 2, 'name': 'akash', 'place': 'machilipatnam', 'phonenumber': 4578122356}
{'_id': ObjectId('635ead793af58a47440d2ce8'), 'telid': 3, 'name': 'manohar', 'place': 'bhimavaram', 'phonenumber': 4589562358}
{'_id': ObjectId('635ead793af58a47440d2ce9'), 'telid': 4, 'name': 'ajay', 'place': 'tanuku', 'phonenumber': 9856231245}
{'_id': ObjectId('635ead793af58a47440d2ceb'), 'telid': 6, 'name': 'lakshmi', 'place': 'nellore', 'phonenumber': 8745568585}
{'_id': ObjectId('635ead793af58a47440d2cec'), 'telid': 7, 'name': 'narayana', 'place': 'nuzvid', 'phonenumber': 7812456235}
{'_id': ObjectId('635ead793af58a47440d2ced'), 'telid': 8, 'name': 'RaghuRamRaavi', 'place': 'moturu', 'phonenumber': 7845962158}
{'_id': ObjectId('635ead793af58a47440d2cee'), 'telid': 9, 'name': 'lakshman', 'place': 'tarigoppula', 'phonenumber': 78954862

In [12]:
data=db.collection1
data.count_documents({})

10

In [13]:
db.collection1.drop() ## dropping the data such that we can use the same for the upcoming one.

# **Student Database (MongoDB)**

Here is the student dataset in the json format. .
Perform the following operation:
First create a database and then load the student.json dataset.
Insert the students record into the collection.
Queries need to answer:

1) Find the student name who scored maximum scores in all (exam, quiz and homework)?

2) Find students who scored below average in the exam and pass mark is 40%?

3) Find students who scored below pass mark and assigned them as fail, and above pass mark as pass in all the categories.

4) Find the total and average of the exam, quiz and homework and store them in a separate collection.

5) Create a new collection which consists of students who scored below average and above 40% in all the categories.

6) Create a new collection which consists of students who scored below the fail mark in all the categories.

7) Create a new collection which\ consists of students who scored above pass mark in all the categories.

Data set Link: https://drive.google.com/file/d/1m7bmdkn6uFCYmo_6zf8VNafqaE2c0FHY/view?usp=sharing


In [14]:
# creating the database for uploading the current data

db = myclient["StudentData"]
data = db.studentcollection
data.count_documents({})

200

In [15]:
data.drop()

In [16]:
student_data = []
for line in open('students.json','r'):
    student_data.append(json.loads(line))

In [17]:
data.drop()

In [18]:
x = data.insert_many(student_data)
print(x.inserted_ids)

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199]


In [19]:
data.count_documents({})

200

In [20]:
data.find_one()

{'_id': 0,
 'name': 'aimee Zank',
 'scores': [{'score': 1.463179736705023, 'type': 'exam'},
  {'score': 11.78273309957772, 'type': 'quiz'},
  {'score': 35.8740349954354, 'type': 'homework'}]}

In [21]:
studentdata = db.studentcollection.find()

for i in studentdata:
    print(i)

{'_id': 0, 'name': 'aimee Zank', 'scores': [{'score': 1.463179736705023, 'type': 'exam'}, {'score': 11.78273309957772, 'type': 'quiz'}, {'score': 35.8740349954354, 'type': 'homework'}]}
{'_id': 1, 'name': 'Aurelia Menendez', 'scores': [{'score': 60.06045071030959, 'type': 'exam'}, {'score': 52.79790691903873, 'type': 'quiz'}, {'score': 71.76133439165544, 'type': 'homework'}]}
{'_id': 2, 'name': 'Corliss Zuk', 'scores': [{'score': 67.03077096065002, 'type': 'exam'}, {'score': 6.301851677835235, 'type': 'quiz'}, {'score': 66.28344683278382, 'type': 'homework'}]}
{'_id': 3, 'name': 'Bao Ziglar', 'scores': [{'score': 71.64343899778332, 'type': 'exam'}, {'score': 24.80221293650313, 'type': 'quiz'}, {'score': 42.26147058804812, 'type': 'homework'}]}
{'_id': 4, 'name': 'Zachary Langlais', 'scores': [{'score': 78.68385091304332, 'type': 'exam'}, {'score': 90.2963101368042, 'type': 'quiz'}, {'score': 34.41620148042529, 'type': 'homework'}]}
{'_id': 5, 'name': 'Wilburn Spiess', 'scores': [{'scor

**1) Find the student name who scored maximum scores in all (exam, quiz and homework)?**

In [22]:
db = myclient.StudentData ## creating a separate collection
ex = db.exam_collection
ex.count_documents({})

200

In [23]:
ex.drop()

In [24]:
for j in range(200):
  x=data.aggregate([{"$match":{"_id":j}},{"$unwind":"$scores"},{"$match":{"scores.type":"exam"}},{"$sort":{"scores.score":-1}}])
  for i in x:
    i=ex.insert_many([i])

In [25]:
#student who score highest score in exam
s=ex.find().sort('scores.score',-1).limit(1)
for i in s:
  print(i)

{'_id': 136, 'name': 'Margart Vitello', 'scores': {'score': 99.33685767140612, 'type': 'exam'}}


In [26]:
ex.count_documents({})

200

In [27]:
db = myclient.StudentData # creating each different collection
qz=db.qzcollection
qz.count_documents({})

200

In [82]:
qz.drop()

In [124]:
for j in range(200):
  x = data.aggregate([{"$match":{"_id":j}},{"$unwind":"$scores"},{"$match":{"scores.type":"quiz"}},{"$sort":{"scores.score":-1}}])
  for i in x:
    i=qz.insert_many([i])

BulkWriteError: batch op errors occurred, full error: {'writeErrors': [{'index': 0, 'code': 11000, 'keyPattern': {'_id': 1}, 'keyValue': {'_id': 0}, 'errmsg': 'E11000 duplicate key error collection: StudentData.qzcollection index: _id_ dup key: { _id: 0 }', 'op': {'_id': 0, 'name': 'aimee Zank', 'scores': {'score': 11.78273309957772, 'type': 'quiz'}}}], 'writeConcernErrors': [], 'nInserted': 0, 'nUpserted': 0, 'nMatched': 0, 'nModified': 0, 'nRemoved': 0, 'upserted': []}

In [125]:
#student who score highest score in quiz
x = qz.find().sort('scores.score',-1).limit(1)
for i in x:
    print(i)

{'_id': 69, 'name': 'Cody Strouth', 'scores': {'score': 99.80348240553108, 'type': 'quiz'}}


In [126]:
qz.count_documents({}) 
## verifying the no.of. documents

200

In [127]:
db = myclient.StudentData
hw=db.hwcollection
hw.count_documents({})

200

In [128]:
hw.drop()

In [129]:
for j in range(200):
  x = data.aggregate([{"$match":{"_id":j}},{"$unwind":"$scores"},{"$match":{"scores.type":"homework"}},{"$sort":{"scores.score":-1}}])
  for i in x:
    i=hw.insert_many([i])

In [130]:
#student who score highest score in homework
x = hw.find().sort('scores.score',-1).limit(1)
for i in x:
    print(i)

{'_id': 178, 'name': 'Whitley Fears', 'scores': {'score': 99.77237745070993, 'type': 'homework'}}


In [131]:
hw.count_documents({})

200

### 1) Find the student name who scored maximum scores in all (exam, quiz and homework)?

#### max score in exam
{'_id': 136, 'name': 'Margart Vitello', 'scores': {'score': 99.33685767140612, 'type': 'exam'}}
#### max score in quiz ---  'Cody Strouth'
{'_id': 69, 'name': 'Cody Strouth', 'scores': {'score': 99.80348240553108, 'type': 'quiz'}} 
#### max score in homework --- 'Whitley Fears'
{'_id': 178, 'name': 'Whitley Fears', 'scores': {'score': 99.77237745070993, 'type': 'homework'}}

**2) Find students who scored below average in the exam and pass mark is 40%?**

In [91]:
# avg score in exam
x = db.excollection.aggregate([{'$group' : {'_id' : "null", "avg_exam" : {'$avg' : "$scores.score"}}}])
for i in x:
    print(i)

In [92]:
db = myclient.StudentData
less =db.lesscollection
less.count_documents({})

0

In [93]:
less.drop()

In [94]:
# less than avg score in exam
x=db.exam_collection.find({'scores.score':{'$lt':48.67367075950175} })
for i in x:
   i=less.insert_many([i])
    #print(i)

In [95]:
less.count_documents({})

104

### 2) Find students who scored below average in the exam and pass mark is 40%? 

lessavg collection shows the students who scored below average in exam is 104

**3) Find students who scored below pass mark and assigned them as fail, and above pass mark as pass in all the categories.**

In [96]:
# 1 Exam Failure List
db = myclient.StudentData
fex=db.fexm
fex.count_documents({})

0

In [97]:
fex.drop()

In [98]:
x=db.exam_collection.find(
    {'scores.score':{'$lt':40}}
)
for i in x:
  i=fex.insert_many([i])

fex.count_documents({})

81

In [99]:
db = myclient.StudentData
pex=db.passex
pex.count_documents({})

0

In [147]:
pex.drop()

In [148]:
x=db.exam_collection.find(
    {'scores.score':{'$gte':40}
     }
)
for i in x:
  i=pex.insert_many([i])

In [149]:
pex.count_documents({})

119

In [56]:
# 2) QUIZ List

In [150]:
db = myclient.StudentData
fqz=db.failqz
fqz.count_documents({})

0

In [151]:
fqz.drop()

In [157]:
x=db.quiz_collection.find(
    {'scores.score':{'$lt':40}
     }
)
for i in x:
  i=fqz.insert_many([i])


In [158]:
db = myclient.StudentData
pqz=db.pass_quiz
pqz.count_documents({})

0

In [159]:
pqz.drop()

In [160]:
x=db.quiz_collection.find(
    {'scores.score':{'$gte':40}}
)
for i in x:
  i=pqz.insert_many([i])

In [161]:
pqz.count_documents({})

0

In [162]:
# Home Work List

In [163]:
db = myclient.StudentData
fhw=db.fail_homework
fhw.count_documents({})

0

In [164]:
fhw.drop()

In [165]:
x=db.homework_collection.find(
    {'scores.score':{'$lt':40}
     }
)
for i in x:
  i=fhw.insert_many([i])
  #print(i)

In [166]:
fhw.count_documents({})

0

In [167]:
db = myclient.StudentData
passhomework=db.pass_homework
passhomework.count_documents({})

0

In [168]:
passhomework.drop()

In [169]:
x=db.hw.find(
    {'scores.score':{'$gte':40}
     }
)
for i in x:
  i=passhomework.insert_many([i])
passhomework.count_documents({})

0

**4) Find the total and average of the exam, quiz and homework and store them in a separate collection.**

In [72]:
# 1) EXAM

In [170]:
db = myclient.StudentData
exam1=db.exam
exam1.count_documents({})

0

In [171]:
exam1.drop()

In [172]:
x= db.ex.aggregate([{"$match": {"scores.type":"exam"}},{ "$unwind":"$scores"},{"$match": { "scores.type":"exam" } },{"$group": {
       "_id": 'total_score_exam',
       "total_score": { "$sum": "$scores.score" } }}])
for i in x:
  i=exam1.insert_many([i])
  print(i)

In [175]:
x=db.exam_collection.aggregate([{'$group' : {'_id' : "avg_score_exam", "avg_score" : {'$avg' : "$scores.score"}}}])
for i in x:
  i=exam1.insert_many([i])
  print(i)

In [176]:
exam1.count_documents({})

1

In [177]:
# 2) QUIZ

In [178]:
db = myclient.StudentData
quiz1=db.quiz
quiz1.count_documents({})

0

In [179]:
quiz1.drop()

In [180]:
x= db.quiz_collection.aggregate([{"$match": {"scores.type":"quiz"}},{ "$unwind":"$scores"},{"$match": { "scores.type":"quiz" } },{"$group": {
       "_id": 'total_score_quiz',
       "total_score": { "$sum": "$scores.score" } }}])
for i in x:
  i=quiz1.insert_many([i])
  print(i)

In [181]:
x=db.quiz_collection.aggregate([{'$group' : {'_id' : "avg_score_quiz", "avg_score" : {'$avg' : "$scores.score"}}}])
for i in x:
  i=quiz1.insert_many([i])
  print(i)

In [182]:
quiz1.count_documents({})

0

In [183]:
# 3) HOMEWORK

In [212]:
db = myclient.StudentData
homework1=db.homework
homework1.count_documents({})

0

In [213]:
homework1.drop()

In [214]:
x= db.homework_collections.aggregate([{"$match": {"scores.type":"homework"}},{ "$unwind":"$scores"},{"$match": { "scores.type":"homework" } },{"$group": {
       "_id": 'total_score_homework',
       "total_score": { "$sum": "$scores.score" } }}])
for i in x:
  i=homework1.insert_many([i])
  print(i)

In [215]:
x=db.homework_collections.aggregate([{'$group' : {'_id' : "avg_score_homework", "avg_score" : {'$avg' : "$scores.score"}}}])
for i in x:
  i=homework1.insert_many([i])
  print(i)

In [216]:
homework1.count_documents({})

0

**5) Create a new collection which consists of students who scored below average and above 40% in all the categories.**

In [ ]:
# 1) EXAM

In [217]:
db = myclient.StudentData
gtltexam=db.gt40_ltavg_exam
gtltexam.count_documents({})

23

In [218]:
gtltexam.drop()

In [219]:
x=db.exam_collection.find({'scores.score':{'$gte':40,'$lt':48.67367075950175}})
for i in x:
  i=gtltexam.insert_many([i])

In [220]:
gtltexam.count_documents({})

23

In [221]:
# 2) QUIZ

In [223]:
db = myclient.StudentData
gtltquiz=db.gt40_ltavg_quiz
gtltquiz.count_documents({})

0

In [224]:
gtltquiz.drop()

In [227]:
x=db.quiz_collections.find({'scores.score':{'$gte':40,'$lt':48.67367075950175}})
for i in x:
  i=gtltquiz.insert_many([i])
  #print(i)

In [228]:
gtltquiz.count_documents({})

0

In [ ]:
# 3) HOME WORK

In [229]:
db = myclient.StudentData
gtlthomework=db.gt40_ltavg_homework
gtlthomework.count_documents({})

0

In [230]:
gtlthomework.drop()

In [231]:
x=db.homework_collections.find({'scores.score':{'$gte':40,'$lt':48.67367075950175}})
for i in x:
  i=gtlthomework.insert_many([i])

In [232]:
gtlthomework.count_documents({})

0


**6) Create a new collection which consists of students who scored below the fail mark in all the categories.**


In [233]:
# 1) EXAM

In [236]:
db = myclient.StudentData
ltfailexam=db.lt_fail_exam
ltfailexam.count_documents({})

0

In [237]:
ltfailexam.drop()

In [238]:
x=db.exam_collection.find({'scores.score':{'$lt':1}})
for i in x:
  i=ltfailexam.insert_many([i])
  #print(i)

In [239]:
ltfailexam.count_documents({})

3

In [240]:
# 2) QUIZ

In [242]:
db = myclient.StudentData
ltfailquiz=db.lt_fail_quiz
ltfailquiz.count_documents({})

0

In [243]:
ltfailquiz.drop()

In [244]:
x=db.quiz_collection.find({'scores.score':{'$lt':1}})
for i in x:
  i=ltfailquiz.insert_many([i])

In [245]:
ltfailquiz.count_documents({})

0

In [246]:
# 3) HOME WORK

In [247]:
db = myclient.StudentData
ltfailhomework=db.lt_fail_homework
ltfailhomework.count_documents({})

0

In [248]:
ltfailhomework.drop()

In [249]:
x=db.homework_collections.find({'scores.score':{'$lt':1}})
for i in x:
  i=ltfailhomework.insert_many([i])
  #print(i)

In [250]:
ltfailhomework.count_documents({})

0


**7) Create a new collection which consists of students who scored above pass mark in all the categories.**

In [251]:
# 1) EXAM

In [252]:
db = myclient.StudentData
gtpassexam=db.gt_pass_exam
gtpassexam.count_documents({})

0

In [253]:
gtpassexam.drop()

In [255]:
x=db.exam_collection.find({'scores.score':{'$gte':40}})
for i in x:
  i=gtpassexam.insert_many([i])

In [256]:
gtpassexam.count_documents({})

119

In [257]:
# 2) QUIZ

In [259]:
db = myclient.StudentData
gtpassquiz=db.gt_pass_quiz
gtpassquiz.count_documents({})

0

In [260]:
gtpassquiz.drop()

In [261]:
x=db.quiz_collection.find({'scores.score':{'$gte':40}})
for i in x:
  i=gtpassquiz.insert_many([i])

In [262]:
gtpassquiz.count_documents({})

0

In [263]:
# 3) HOME WORK

In [265]:
db = myclient.StudentData
gtpasshomework=db.gt_pass_homework
gtpasshomework.count_documents({})

0

In [266]:
gtpasshomework.drop()

In [271]:
x=db.homework_collections.find({'scores.score':{'$gte':40}})
for i in x:
  i=gtpasshomework.insert_many([i])

In [272]:
gtpasshomework.count_documents({})

0